src/sentry/static/sentry/app/views/discover/index.tsx

```diff
@@ -204,11 +201,15 @@ class DiscoverContainer extends React.Component<Props, State> {
     });
   };
 
-  renderNoAccess() {
-    return (
-      <AlertAsAny type="warning">{t("You don't have access to this feature")}</AlertAsAny>
-    );
-  }
+  renderNoAccess = () => {
+    const {router, organization} = this.props;
+
+    const route = organization.features.includes('discover-query')
+      ? `/organizations/${organization.slug}/discover/queries/`
+      : `/organizations/${organization.slug}/discover/results/`;
```

Comment:
@Zylphrex I think if the org has neither `discover-query` nor `discover-basic`, we can still show the "You don't have access to this feature" banner. See the percy snapshots for this: https://github.com/getsentry/sentry/blob/26f283d4ec5d16978952e78ca3673eebe96e65db/tests/acceptance/test_organization_discover.py#L55

This should gracefully fallback to the same behaviour prior to this PR for onpremise folks.

Reply:
To confirm, when an org does not have `discover-basic` nor `discover-query`, they should be kept at the discover v1 landing page and be presented with the no access to features alert?

@dashed The failing percy snapshot is due to the fact that the test org is not given `discover-page` but does have `discover-query` thus it redirects tho the queries page.

Reply:
I'm unsure what `discover-page` is. But I think that sounds right. 

We should expect the percy snapshot to not change.

Reply:
> To confirm, when an org does not have discover-basic nor discover-query, they should be kept at the discover v1 landing page and be presented with the no access to features alert?

Sending the user to discover2 when they lack both `discover` and `discover-basic` should still result in a 'no access' view, which is what I'm seeing in the latest percy diffs.